src/core/event_registry/server_event.rs

```diff
@@ -750,6 +704,133 @@ impl SerializedMessage {
     fn event_bytes(&self) -> &[u8] {
         &self.bytes[self.tick_size..]
     }
+
+    /// Optimized to avoid reallocations when clients have the same init tick as the server's latest init tick (at
+    /// the time the event was submitted).
+    fn get_bytes(&self, init_tick: RepliconTick) -> bincode::Result<Bytes> {
+        if self.init_tick == init_tick {
```

Comment:
If init tick changes it will result in reallocation for all events.
Maybe serialize as `Vec` and then on sending convert into `Bytes`, storing the last sent message like we did before in the old `send_event` function? This should result in less reallocs.

This way you can also just serialize zeroes instead of the tick equal to the number of bytes of `ServerTick`. And update it  on first sending to a client anyway. Just to avoid tracking `ServerLastInitTick`.